From f6ff93f7b7db154b7899f415ec67677e6ddd7ff5 Mon Sep 17 00:00:00 2001
From: Victor Berger <victor.berger@m4x.org>
Date: Thu, 8 Jul 2021 19:39:30 +0200
Subject: [PATCH] Fix build when using GTest-1.11

Some change in GTest 1.11 headers caused a conflict on the name
`Pointer`. Removing the unconditional import of namespace `testing`
in favor of individual imports of the used items fixes it.

cc #205
---
 tests/pointer_constraints.cpp | 6 +++++-
 tests/relative_pointer.cpp    | 6 +++++-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/tests/pointer_constraints.cpp b/tests/pointer_constraints.cpp
index d306710..6ee856f 100644
--- a/tests/pointer_constraints.cpp
+++ b/tests/pointer_constraints.cpp
@@ -24,7 +24,11 @@
 
 #include <memory>
 
-using namespace testing;
+using testing::AnyNumber;
+using testing::Eq;
+using testing::Ne;
+using testing::NotNull;
+
 using namespace wlcs;
 
 namespace
diff --git a/tests/relative_pointer.cpp b/tests/relative_pointer.cpp
index c33316c..c5c5358 100644
--- a/tests/relative_pointer.cpp
+++ b/tests/relative_pointer.cpp
@@ -22,7 +22,11 @@
 
 #include <gmock/gmock.h>
 
-using namespace testing;
+using testing::AnyNumber;
+using testing::IsTrue;
+using testing::NotNull;
+using testing::_;
+
 using namespace wlcs;
 
 namespace
-- 
2.32.0

